shipping/uc_usps/uc_usps.admin.inc

```diff
@@ -256,15 +405,44 @@ function uc_usps_admin_settings_submit($form, &$form_state) {
 
   foreach ($fields as $key) {
     $value = $form_state['values'][$key];
-
     if (is_array($value) && isset($form_state['values']['array_filter'])) {
       $value = array_keys(array_filter($value));
     }
-    config_set('uc_usps.settings', $key, $value);
+    $config->set($key, $value);
   }
 
+  // Update OAuth2 client secret if we entered a value.
+  $client_secret = $form_state['values']['uc_usps_client_secret'];
+  if (!empty($client_secret)) {
+    $config->set('uc_usps_client_secret', $client_secret);
+  }
+  // Save all config changes now because _uc_usps_fetch_access_token() will also
+  // modify the config file.
+  $config->save();
   backdrop_set_message(t('The configuration options have been saved.'));
 
+  // Check token expiration if we're using the new API.
+  if ($form_state['values']['uc_usps_authorization_method'] == 'newapi') {
+    $token_expiry = $config->get('uc_usps_token_expiry');
```

Comment:
Token expiry and anything else that gets automatically updated by the API service should be stored in `state` (with `state_set()` and `state_get()`). Otherwise config will often be out of sync between active and staging, and someone would potentially wipe out the token in the active config if importing staged config.

Reply:
Good point. I"ll make that change.